lib/libzpool/taskq.c

```diff
@@ -241,6 +244,16 @@ taskq_thread(void *arg)
 		if (!prealloc)
 			task_free(tq, t);
 	}
+	/*
+	 * This thread is on its way out, so just drop the lock temporarily
+	 * in order to call the shutdown callback.  This allows the callback
+	 * to look at the taskqueue, even just before it dies.
+	 */
+	if (tq->tq_dtor != NULL) {
+		mutex_exit(&tq->tq_lock);
+		tq->tq_dtor(tq);
+		mutex_enter(&tq->tq_lock);
+	}
```

Comment:
Is there a reason, since this is all new code and the dtor callback would be new, to not simply have it deal with any locking if it needs to?